src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,83 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { useDispatch, useSelector } from 'react-redux';
+import { RootState } from '../../app/store';
+import { clearCurrentTodo } from '../../features/currentTodo';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
 
 export const TodoModal: React.FC = () => {
+  const dispatch = useDispatch();
+  const currentTodo = useSelector((state: RootState) => state.currentTodo);
+  const { id, title, completed, userId } = currentTodo!;
+  const [userInfo, setUserInfo] = useState<User | null>(null);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+
+  useEffect(() => {
+    const findUser = async (newUser: number) => {
+      setIsLoading(true);
+      try {
+        const user = await getUser(newUser);
+
+        setUserInfo(user);
+      } catch {
+        dispatch(clearCurrentTodo());
+        alert(`Didn't find user info`);
+      } finally {
+        setIsLoading(false);
+      }
+    };
+
+    findUser(userId);
```

Comment:
It is not safe to directly use `userId` inside the `useEffect` hook without checking if it is not null or undefined. It might cause unexpected behavior or even crash your application.